week-3/Banking/Banking.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.17;
+
+contract Banking {
+    struct Account {
+        address payable walletAddress;
+        uint256 balance;
+    }
+
+    // Banks customer database
+    mapping(address => Account) internal recordBook;
+
+    function createAccount() public {
+        // Check if user is already registered in Bank's recordbook or not
+        require(
+            recordBook[msg.sender].walletAddress == address(0),
+            "Account Already Exist!!"
+        );
+
+        // Create account for user with initial values
+        recordBook[msg.sender] = Account(payable(msg.sender), 0);
+        return;
+    }
+
+    function deposite() public payable userExist {
+        recordBook[msg.sender].balance =
+            msg.value +
+            recordBook[msg.sender].balance;
+        return;
+    }
+
+    function withdraw(
+        uint256 _amount
+    ) public userExist checkFundAvailability(_amount) {
+        // make address poayable and pay to customer
+        address payable customer = recordBook[msg.sender].walletAddress;
+        customer.transfer(_amount);
```

Comment:
Update the balance of the account before transferring. Explore "Reentrancy Hack".